rest-api/tools/setup_database.sh

```diff
@@ -100,36 +91,51 @@ function finish {
 }
 trap finish EXIT
 
-echo '{"db_connection_string": "'$CONNECTION_STRING'", ' \
-     ' "backup_db_connection_string": "'$BACKUP_CONNECTION_STRING'", '\
-     ' "rdr_db_password": "'$RDR_PASSWORD'", ' \
-     ' "root_db_password": "'$ROOT_PASSWORD'", ' \
-     ' "read_only_db_password": "'$READONLY_PASSWORD'", ' \
-     ' "db_connection_name": "'$INSTANCE_CONNECTION_NAME'", '\
-     ' "backup_db_connection_name": "'$BACKUP_INSTANCE_NAME'", '\
-     ' "db_user": "'$RDR_DB_USER'", '\
-     ' "db_name": "'$DB_NAME'" }' > $TMP_DB_INFO_FILE
-
-for db_name in "rdr" "metrics"; do
-  if [ "${UPDATE_PASSWORDS}" = "Y" ]
-  then
-    cat tools/update_passwords.sql | envsubst > $UPDATE_DB_FILE
-  else
-    cat tools/create_db.sql | envsubst > $UPDATE_DB_FILE
-  fi
-
-  run_cloud_sql_proxy
-
-  if [ "${UPDATE_PASSWORDS}" = "Y" ]
+if [ "${UPDATE_PASSWORDS}" = "Y" ] || [ "${CREATE_INSTANCE}" = "Y" ]
   then
-    echo "Updating database user passwords..."
+    	echo "Updating database user passwords..."
+	randpw
+	ROOT_PASSWORD=$new_password
+	randpw
+	RDR_PASSWORD=$new_password
+	randpw
+	READONLY_PASSWORD=$new_password
+
+	echo '{"db_connection_string": "'$CONNECTION_STRING'", ' \
+	     ' "backup_db_connection_string": "'$BACKUP_CONNECTION_STRING'", '\
+	     ' "rdr_db_password": "'$RDR_PASSWORD'", ' \
+	     ' "root_db_password": "'$ROOT_PASSWORD'", ' \
+	     ' "read_only_db_password": "'$READONLY_PASSWORD'", ' \
+	     ' "db_connection_name": "'$INSTANCE_CONNECTION_NAME'", '\
+	     ' "backup_db_connection_name": "'$BACKUP_INSTANCE_NAME'", '\
+	     ' "db_user": "'$RDR_DB_USER'", '\
+	     ' "db_name": "'$DB_NAME'" }' > $TMP_DB_INFO_FILE
+
+	echo "Setting root password..."
+	#gcloud sql instances set-root-password $INSTANCE_NAME --password $ROOT_PASSWORD
+	gcloud sql users set-password root % --instance $INSTANCE_NAME --password $ROOT_PASSWORD
+
+	if [ "${UPDATE_PASSWORDS}" = "Y" ]
+	    then 
+		echo "updating passwords for database"
+		cat tools/update_passwords.sql | envsubst > $UPDATE_DB_FILE
+	else
+		echo "creating new database"
+		for db_name in "rdr" "metrics"; do
+		   cat tools/create_db.sql | envsubst > $UPDATE_DB_FILE
+		   cat tools/grant_permissions.sql | envsubst >> $UPDATE_DB_FILE
+		done
+	fi
+
+	mysql -u "$ROOT_DB_USER" -p"$ROOT_PASSWORD" --host 127.0.0.1 --port ${PORT} < ${UPDATE_DB_FILE}
   else
-    echo "Creating empty database..."
-  fi
-  mysql -u "$ROOT_DB_USER" -p"$ROOT_PASSWORD" --host 127.0.0.1 --port ${PORT} < ${UPDATE_DB_FILE}
-done
+	echo "getting database config..."
+	for db_name in "rdr" "metrics"; do
+	   cat tools/grant_permissions.sql | envsubst > $UPDATE_DB_FILE
+	done
+fi
 
 echo "Setting database configuration..."
```

Comment:
This should only run in the first half of the if statement... we don't want to update the db_config when just granting permissions.

Reply:
done